src/childrenHavePropXorChildren.js

```diff
@@ -7,22 +7,27 @@ export default function childrenHavePropXorChildren(prop) {
   }
 
   const validator = function childrenHavePropXorChildrenWithProp({ children }, _, componentName) {
-    const childrenCount = React.Children.count(children);
+    let nonFalseyChildrenCount = 0;
     let propCount = 0;
     let grandchildrenCount = 0;
 
     React.Children.forEach(children, (child) => {
-      if (child.props[prop]) {
-        propCount += 1;
-      }
-      if (React.Children.count(child.props.children)) {
-        grandchildrenCount += 1;
+      if (child) {
```

Comment:
i think we can avoid the nesting by using an early return - ie, `if (!child) { return; }`